datajunction-server/datajunction_server/database/collection.py

```diff
@@ -0,0 +1,79 @@
+"""Collection database schema."""
+from datetime import datetime, timezone
+from functools import partial
+from typing import List, Optional
+
+from sqlalchemy import DateTime, ForeignKey, String, select
+from sqlalchemy.ext.asyncio import AsyncSession
+from sqlalchemy.orm import Mapped, mapped_column, relationship
+
+from datajunction_server.database.base import Base
+from datajunction_server.database.node import Node
+from datajunction_server.errors import DJDoesNotExistException
+from datajunction_server.typing import UTCDatetime
+
+
+class Collection(Base):  # pylint: disable=too-few-public-methods
+    """
+    A collection of nodes
+    """
+
+    __tablename__ = "collection"
+
+    id: Mapped[int] = mapped_column(primary_key=True)
+    name: Mapped[Optional[str]] = mapped_column(String, default=None, unique=True)
+    description: Mapped[Optional[str]] = mapped_column(String, default=None)
+    nodes: Mapped[List[Node]] = relationship(
```

Comment:
Since we store a list of Nodes in here, why would we need a separate `collectionnodes` table?

Reply:
`collectionnodes` is the association table, since each collection can have many nodes and each node can have many collections.